src/storage/index.ts

```diff
@@ -27,6 +27,8 @@ import { NAME_LOOKUP_PATH } from '../auth/authConstants'
 import { getGlobalObject, getBlockstackErrorFromResponse, megabytesToBytes } from '../utils'
 import { fetchPrivate } from '../fetchUtil'
 
+const etags: { [key: string]: string; } = {}
```

Comment:
I think having this as a global variable is okay for now, but we should create an issue for moving this into `UserSession.store` (an instead of the `SessionDataStore` class). 

This would allow the data store to persist this map across sessions. As well fix duplicate etag-file-key bugs with certain applications that may be working with multiple `UserSession` instances. 

We will have to complete https://github.com/blockstack/blockstack.js/issues/744 before being able to implement that though.